bridge/evm/test/SuiBridgeTest.t.sol

```diff
@@ -867,3 +880,24 @@ contract SuiBridgeTest is BridgeBaseTest, ISuiBridge {
         assertEq(_calldata, hex"5cd8a76b");
     }
 }
+
+contract ReentrantAttack {
+    IBridgeVault public vault;
+    bool private attackInitiated;
+
+    constructor(address _vault) {
+        vault = IBridgeVault(_vault);
+    }
+
+    receive() external payable {
+        if (!attackInitiated) {
+            attackInitiated = true;
+            vault.transferETH(payable(address(this)), 100);
+        }
+    }
+
+    function attack() external payable {
+        attackInitiated = false;
+        vault.transferETH(payable(address(this)), 100);
+    }
```

Comment:
why not 
```
    receive() external payable {
        if (!attackInitiated) {
            vault.transferETH(payable(address(this)), 100);
        }
    }

    function attack() external payable {
        vault.transferETH(payable(address(this)), 100);
    }
```

Reply:
This would cause an Infinite loop / any call to the attack() function would run out of gas.

Reply:
wouldn't the reentrancy guard catch this in the 2nd loop?